contracts.go

```diff
@@ -32,6 +32,8 @@ type Provider interface {
 
 	Exec(ctx context.Context, query string, params ...interface{}) (Result, error)
 	Transaction(ctx context.Context, fn func(Provider) error) error
+
+	Close() error
```

Comment:
So great job, and thanks for reading and understanding my project, I appreciate it =].

But about this particular addition (which forces us to add more tests and modify the mocks), I would expect this not to be necessary, since the `kpgx.New()` functions return a `ksql.DB` instance so even without modifying the interface we could still do something like this:

```golang
db, err := kpgx.New(ctx, os.Getenv("PG_URL"), ksql.Config{})
if err != nil { ... }
defer db.Close()
```

Which is what I think we need, is there something I am missing regarding the need to update this interface?

Regards, Vinícius Garcia.

Reply:
Hi, this make sense. I was thinking that it should be an interface constraint, :)